### PR TITLE
Add documentation on how to debug the kernel of a VM

### PR DESCRIPTION
**Pull request by dthaler**
Signed-off-by: Dave Thaler <dthaler@microsoft.com>

### CODE REVIEW

**Review by Alan-Jowett — APPROVED** [review]

File: docs/vm-setup.md [hunk 1]
```diff
@@ -40,3 +40,20 @@ has already built the binaries for x64/Debug.
...
+
+To use Windbg or KD as the debugger instead of Visual Studio, see [Setting Up Kernel-Mode Debugging of a Virtual Machine Manually using a Virtual COM Port](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/attaching-to-a-virtual-machine--kernel-mode-)
+
+Both of the links above contain the statement:
+"In the virtual machine, configure the COM port to map to a named pipe. The debugger will connect through this pipe.
+For more information about how to create this pipe, see your virtual machine's documentation."
+
+That step can be done from the *Host* as follows:
+
+1. Start an admin powershell.
```
Comment (shankarseal) [inline docs/vm-setup.md:56]: For gen2 VMs, I think secure boot has to be turned off first.

Comment (shankarseal) [reply docs/vm-setup.md:56]: Also please mention this as an alternative (and even prefer it IMO)
https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/setting-up-network-debugging-of-a-virtual-machine-host

**Review by shankarseal — APPROVED** [review]

File: docs/vm-setup.md (hunk 1 above)
Comment (dthaler) [reply docs/vm-setup.md:56]: > For gen2 VMs, I think secure boot has to be turned off first. Set-VMFirmware –Vmname VmName –EnableSecureBoot Off

It's already there at line 19 of this file.

Comment (dthaler) [reply docs/vm-setup.md:56]: > Also please mention this as an alternative (and even prefer it IMO) https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/setting-up-network-debugging-of-a-virtual-machine-host

I will add it as an alternative, but that link does not contain the instructions on how to use it from VS, unlike the other links already in here.  So we'd need to dig out an additional link, or add the instructions here.

**Pull request merged**